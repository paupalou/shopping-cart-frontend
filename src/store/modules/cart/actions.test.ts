import {
  addProductsToCartRequestAction,
  addProductsToCartSuccessAction,
  addProductsToCartFailureAction
} from './actions';

describe('Cart actions', () => {
  it('Add products to cart request action', () => {
    const payload = [
      { id: 1, quantity: 2 },
      { id: 4, quantity: 5 }
    ];
    expect(addProductsToCartRequestAction(payload)).toMatchSnapshot();
  });

  it('Add products to cart success action', () => {
    const payload = {
      items: [
        {
          id: 1,
          name: 'Soup',
          quantity: 2,
          price: '$100.00',
          total: '$200.00'
        }
      ],
      loading: false,
      error: false,
      total: '$200.00'
    };

    expect(addProductsToCartSuccessAction(payload)).toMatchSnapshot();
  });

  it('Add products to cart failure action', () => {
    expect(addProductsToCartFailureAction()).toMatchSnapshot();
  });
});
