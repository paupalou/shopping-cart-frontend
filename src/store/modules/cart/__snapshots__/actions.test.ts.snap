// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart actions Add products to cart failure action 1`] = `
Object {
  "type": "ADD_PRODUCTS_TO_CART_FAILURE",
}
`;

exports[`Cart actions Add products to cart request action 1`] = `
Object {
  "payload": Array [
    Object {
      "id": 1,
      "quantity": 2,
    },
    Object {
      "id": 4,
      "quantity": 5,
    },
  ],
  "type": "ADD_PRODUCTS_TO_CART_REQUEST",
}
`;

exports[`Cart actions Add products to cart success action 1`] = `
Object {
  "payload": Object {
    "error": false,
    "items": Array [
      Object {
        "id": 1,
        "name": "Soup",
        "price": "$100.00",
        "quantity": 2,
        "total": "$200.00",
      },
    ],
    "loading": false,
    "total": "$200.00",
  },
  "type": "ADD_PRODUCTS_TO_CART_SUCCESS",
}
`;
